File: SpringBootApp/src/main/java/com/example/springbootapp/data/dto/ProductDto.java
package com.example.springbootapp.data.dto;

import com.example.springbootapp.data.entities.Enums.Gender;
import com.example.springbootapp.data.entities.Enums.ProductCategory;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
public class ProductDto {
    private String id;
    private String name;
    private String description;
    private Double price;
    private String brand;
    private Gender gender;
    private ProductCategory category;
    private List<ProductPicDto> pics;
    private String clubName;
    private List<ProductWithVariantAvailabilityDto> variants;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/dto/ProductDto.kt
package com.example.clientuser.model.dto

import com.example.clientuser.model.enumerator.Gender
import com.example.clientuser.model.enumerator.ProductCategory
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ProductDto (
    val id: String,
    val name: String,
    val description: String,
    val price: Double,
    val brand: String,
    val gender: Gender,
    val category: ProductCategory,
    val pics: List<ProductPicDto>,
    val clubName: String,
    val variants: List<ProductWithVariantAvailabilityDto>
)